### DISCUSSION TITLE
KernelArguments lifecycle

### DISCUSSION THREAD

**Discussion by MithrilMan** [microsoft/semantic-kernel#6180] [Q&A] [answered]
What's the KernelArguments lifecycle, especially when native function are called with AutoInvokeKernelFunctions?
I thought that injecting KernelArguments and add some contextual informations there would be carried out to the next automatically invoked function but... it's not the case.
What's the point then to have KernelArguments injected into functions?

Any plan to support my described scenario?

**Comment by madsbolaris** [comment, marked as answer]
`KernelArguments` are only passed one level deep (they're essentially just our way of providing input params). If you're interested in passing data through nested functions, you'll want to create a new `KernelArgument`s object that's passed to the next function.

If you want a persistent property bag, I'd recommend using `Kernel.Data`

**Reply by MithrilMan** [reply]
I ended up injecting a Scoped class where I put my data and use FromKernelService to get it in my plugin methods, didn't knew about Kernel.Data, basically I replicated that :/
Well, tbh my object has typed properties but good to know, thanks
